util/parsenum.h

```diff
@@ -64,7 +64,9 @@ _Pragma("clang diagnostic pop")
 			((*(x)) = parsenum_float((s), (min), (max))) :	\
 		(((*(x)) = -1) <= 0) ?					\
 			((*(x)) = parsenum_signed((s), (min), (max))) :	\
-			(((*(x)) = parsenum_unsigned((s),		\
+			(assert(((min) >= 0) && "PARSENUM unsigned with negative minimum bound"),			\
+			assert(((max) <= *(x)) && "PARSENUM unsigned upper bound does not fit in datatype"),			\
```

Comment:
I don't think we want this.  It would be entirely reasonable to say "parse a size_t and make sure the value is less than 5 billion"; this assertion would then pass on a 64-bit system but fail on a 32-bit system.

(As for a negative minimum bound... it's hard to see why someone would do that, but I can't imagine someone doing that by accident either.)